packages/drivers/odsp-driver/src/createFile.ts

```diff
@@ -120,6 +121,61 @@ export async function createNewFluidFile(
     return odspResolvedUrl;
 }
 
+/**
+ * If user requested creation of a sharing link along with the creation of the file by providing either
+ * createLinkType (now deprecated) or createLinkScope in the request parameters, extract and save
+ * sharing link information from the response if it is available.
+ * In case there was an error in creation of the sharing link, error is provided back in the response,
+ * and does not impact the creation of file in ODSP.
+ * @param requestedSharingLinkKind - Kind of sharing link requested to be created along with the creation of file.
+ * @param response - Response object received from the /snapshot api call
+ * @returns Sharing link information received in the response from a successful creation of a file.
+ */
+function extractShareLinkData(
+    requestedSharingLinkKind: ShareLinkTypes | ISharingLinkKind | undefined,
+    response: ICreateFileResponse,
+    enableSingleRequestForShareLinkWithCreate?: boolean,
+    enableShareLinkWithCreate?: boolean,
+): ShareLinkInfoType | undefined {
+    if (!requestedSharingLinkKind) {
+        return;
+    }
+    let shareLinkInfo: ShareLinkInfoType | undefined;
+    if (enableSingleRequestForShareLinkWithCreate) {
+        const { sharing } = response;
+        if (!sharing) {
+            return;
+        }
+        shareLinkInfo = {
+            createLink: {
+                type: requestedSharingLinkKind,
+                link: sharing.sharingLink ? {
+                    scope: sharing.sharingLink.scope,
+                    role: sharing.sharingLink.type,
+                    webUrl: sharing.sharingLink.webUrl,
+                    ...sharing.sharingLink,
+                } : undefined,
+                error: sharing.error,
+                shareId: sharing.shareId,
+            },
+        };
+    } else if (enableShareLinkWithCreate) {
+        const { sharing, sharingLink, sharingLinkErrorReason } = response;
+        if (!sharingLink && !sharingLinkErrorReason) {
```

Comment:
It should be !(A && B)

Reply:
The purpose here is to return undefined if there neither sharingLink nor sharingLinkErrorReason in the response. So if sharingLink is undefined as well as sharingLinkErrorReason is undefined, the condition with become true only in that case and we will just return undefined.